crates/pallets/starknet/src/lib.rs

```diff
@@ -963,4 +963,31 @@ impl<T: Config> Pallet<T> {
         PendingEvents::<T>::try_append(sn_event.clone()).map_err(|_| EventError::TooManyEvents)?;
         Ok(sn_event)
     }
+
+    /// Estimate the fee associated with transaction
+    pub fn estimate_fee(transaction: Transaction) -> Result<(u64, u64), DispatchError> {
```

Comment:
Can we add explanation on why it's fine to call the regular execute function here without worrying about state changes.